app-backend/tool/src/main/scala/eval/LazyTile.scala

```diff
@@ -286,4 +290,26 @@ object LazyTile {
     def bind(args: Map[Var, LazyTile]): LazyTile =
       this
   }
+
+  case class FocalMax(left: LazyTile, n: Neighborhood, gridbounds: Option[GridBounds]) extends Tree {
+    lazy val intTile = focal.Max(left.evaluate.get, n, gridbounds)
+    lazy val dblTile = focal.Max(left.evaluateDouble.get, n, gridbounds)
```

Comment:
Yes, but it can't be avoided without implementing focal on LazyTile. I think this is a good milestone for focal implementation. Bigger problems are likely waiting to be flushed out in testing.

From previous benchmarking we know that the [cursor](https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/mapalgebra/focal/Cursor.scala) implementation is orders of magnitude or so faster than naive "collect the neighborhood for every pixel" implementation. 

It is possible to have a lazy cursor but it would involve a pretty big code upheaval right now. I would agree with skirting the laziness here.